src/tcgwars/logic/impl/gen8/DarknessAblaze.groovy

```diff
@@ -1339,7 +1411,12 @@ public enum DarknessAblaze implements LogicCardInfo {
         weakness L
         bwAbility "Primal Law", {
           text "If this Pokémon is your Active Pokémon, your opponent can’t play any Pokémon from their hand to evolve their Pokémon."
-          actionA {
+          delayedA {
+            before EVOLVE, {
+              if (self.active && bg.currentTurn == self.owner.opposite && ef.evolutionCard.player.pbg.hand.contains(ef.evolutionCard)) {
+                prevent()
+              }
+            }
```

Comment:
This might need a check on it only covering "played from hand". Looks good besides that.

Reply:
That's what that last condition is. Relevant rulings if you're talking about something like Rare Candy

> Ancient Powerげんしのちから
Each player can't play any Pokémon from his or her hand to evolve his or her Pokémon.

> Q. Does Archeops' "Ancient Power" Ability stop attacks that evolve Pokemon, such as Karrablast's "Mysterious Evolution"?
A. Ancient Power only prevents you from playing Pokemon from your hand to evolve other Pokemon. Mysterious Evolution's effect is not stopped because it evolves the Pokemon from a card in the deck, not a card in your hand. (Noble Victories FAQ; Nov 17, 2011 PUI Rules Team)

> Q. Does Rare Candy override evolution-preventing effects such as Archeops' "Ancient Power" Ability?
A. Ancient Power does not allow you to use a Pokemon's attack or play a Trainer card to evolve your Benched or Active Pokemon, unless the Evolution card you would be putting into play comes from somewhere other than your hand. So Rare Candy cannot be used while Archeops' Ancient Power Ability is in effect. (Noble Victories FAQ; Nov 17, 2011 PUI Rules Team; Nov 21, 2011 PUI Rules Team)

Reply:
Ah, missed the last condition somehow (and I even reread 2 times just to make sure 😅 )